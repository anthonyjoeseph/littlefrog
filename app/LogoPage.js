// app/LogoPage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';

import Button from 'react-native-button'

import { Actions } from 'react-native-router-flux';

const LogoPage = () => {
  return (
    <View style={styles.container}>
      <View style={{flex:0.5, justifyContent: 'flex-end'}}>
        <Text style={styles.broccoliLogo}>broccoli</Text>
        <Text style={styles.title}>Little Frog Doesn&#39;t Listen</Text>
      </View>
      <View style={styles.buttons}>
        <View style={{flexDirection:'row', justifyContent:'center', alignItems:'stretch'}}>
          <View style={{flex:2}}></View>
          <Button
            containerStyle={{padding:5, height:45, overflow:'hidden', borderRadius:40, backgroundColor: 'white'}}
            style={{flex:3, textAlign: 'center', backgroundColor: '#FFFFFF', color: 'red', fontSize:40}}
            onPress={() => Actions.bookReader()}
          >Join us!</Button>
          <View style={{flex:1}}></View>
          <Button
            containerStyle={{padding:5, height:45, overflow:'hidden', borderRadius:40, backgroundColor: 'white'}}
            style={{flex:3, textAlign: 'center', backgroundColor: '#FFFFFF', color: 'red', fontSize:40}}
            onPress={() => Actions.bookReader()}
          >Proceed</Button>
          <View style={{flex:2}}></View>
        </View>
      </View>
    </View>
  );
}

//

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#1586BA',
    flexDirection: 'column',
    height: 100,
    padding: 20
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    color: '#FFFFFF'
  },
  broccoliLogo: {
    fontSize: 40,
    textAlign: 'center',
    color: '#FFFFFF'
  },
  buttons: {
    flex: 0.5,
    justifyContent: 'flex-end',
    alignItems: 'center',
    flexDirection: 'column'
  }
});

export default LogoPage;
