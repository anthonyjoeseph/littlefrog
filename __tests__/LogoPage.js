// __tests__/DismissableFullscreenPopup-test.js
import 'react-native';
import React from 'react';
import LogoPage from '../app/scenes/main/LogoPage/LogoPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <LogoPage />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
