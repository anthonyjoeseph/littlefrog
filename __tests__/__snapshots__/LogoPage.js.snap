exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#1586BA",
      "flex": 1,
      "flexDirection": "column",
      "height": 100,
      "justifyContent": "center",
      "padding": 20,
    }
  }>
  <View
    style={
      Object {
        "flex": 0.5,
        "justifyContent": "flex-end",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 40,
          "textAlign": "center",
        }
      }>
      broccoli
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "textAlign": "center",
        }
      }>
      Little Frog Doesn\'t Listen
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <View
        style={
          Object {
            "flex": 2,
          }
        } />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 40,
            "height": 45,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 5,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 17,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              null,
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "red",
                "flex": 3,
                "fontSize": 40,
                "textAlign": "center",
              },
              null,
            ]
          }>
          Join us!
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        } />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 40,
            "height": 45,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 5,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 17,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              null,
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "red",
                "flex": 3,
                "fontSize": 40,
                "textAlign": "center",
              },
              null,
            ]
          }>
          Proceed
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        } />
    </View>
  </View>
</View>
`;
